### PR TITLE
Remove Base64 value length limit from Utf8JsonWriter

### LINKED ISSUE #39953: The JSON value of length n is too large and not supported #2
I'm referring to this issue https://github.com/dotnet/runtime/issues/30746 that was closed with limit of 125MB staying fixed opposed to being configurable.

It was argued that there would be no common cases hitting the 125MB limit. Such cases probably never occur on UI but there are such cases...for example if you have a REST API that returns JSON document which has some document encoded as a base64 in a single field or it has another JSON document nested inside a single field as a string which is used for digital signing and verification of it. 
And i have such a case where i'm digitally signing some JSON document and the creating a new JSON where the payload field (contains my JSON document) is larger than the 125MB limit.

### PR DESCRIPTION
**Pull request by stephentoub**
Closes https://github.com/dotnet/runtime/issues/39953 (unless we want to use it to track removing the other limits)

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs [hunk 1]
```diff
@@ -70,7 +70,6 @@ internal static partial class JsonConstants
 
         public const int MaxEscapedTokenSize = 1_000_000_000;   // Max size for already escaped value.
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs:71]: Could we get rid of this limit too? FWIW I skimmed through the code and it looks like the recently added `WriteRawValue` methods completely disregard these limits.

Comment (stephentoub) [reply src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs:71]: I'd like to get rid of all of the hardcoded limits.  But let's do them individually.

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs [hunk 2]
```diff
@@ -3118,7 +3119,7 @@ public void WritingTooLargeBase64Bytes(bool formatted, bool skipValidation)
 
             using (var jsonUtf8 = new Utf8JsonWriter(output, options))
             {
-                Assert.Throws<ArgumentException>(() => jsonUtf8.WriteBase64StringValue(value.AsSpan(0, 125_000_001)));
+                jsonUtf8.WriteBase64StringValue(value.AsSpan(0, 125_000_001));
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs:3122]: Should this be asserting on the JSON output now?

File: src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs [hunk 3]
```diff
@@ -87,6 +87,12 @@ public static void ThrowInvalidOperationException_NeedLargerSpan()
             throw GetInvalidOperationException(SR.FailedToGetLargerSpan);
         }
 
+        [DoesNotReturn]
+        public static void ThrowPropertyNameTooLargeArgumentException(int length)
+        {
+            throw GetArgumentException(SR.Format(SR.PropertyNameTooLarge, length));
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs:93]: If the `PropertyNameTooLarge` res string already exists, shouldn't there be a corresponding throw method already? Or was it unused/used differently until now?

**Review by eiriktsarpalis — APPROVED** [review]

File: src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs (hunk 3 above)
Comment (stephentoub) [reply src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs:93]: PropertyNameTooLarge is used in other throw helpers that check multiple arguments, one of which is a property being checked for its length.

**Pull request merged**